pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -62,25 +71,76 @@ def __init__(self, logger, handler, socket_path, platform_kind, namespace=None,
         # indicate that we're ready
         self._write_packet_to_processor('s')
 
+    @staticmethod
+    def _event_from_msgpack(parsed_data):
+        """Decode event encoded as MessagePack by processor"""
+
+        trigger = TriggerInfo(
+            parsed_data['trigger']['class'],
+            parsed_data['trigger']['kind'],
+        )
+
+        # extract content type, needed to decode body
+        content_type = parsed_data['content_type']
+
+        body = nuclio_sdk.Event.decode_body(parsed_data['body'], content_type)
+
+        return nuclio_sdk.Event(body=body,
+                                content_type=content_type,
+                                trigger=trigger,
+                                fields=parsed_data.get('fields'),
+                                headers=parsed_data.get('headers'),
+                                _id=parsed_data['id'],
+                                method=parsed_data['method'],
+                                path=parsed_data['path'],
+                                size=parsed_data['size'],
+                                timestamp=datetime.datetime.utcfromtimestamp(parsed_data['timestamp']),
+                                url=parsed_data['url'],
+                                _type=parsed_data['type'],
+                                type_version=parsed_data['type_version'],
+                                version=parsed_data['version'])
+
     def serve_requests(self, num_requests=None):
         """Read event from socket, send out reply"""
 
+        int_buf = bytearray(8)
+        buf = memoryview(bytearray(4 * 1024 * 1024))
+
         while True:
 
             formatted_exception = None
             encoded_response = '{}'
 
             try:
-                line = self._processor_sock_rfile.readline()
 
+                should_be_eight = self._processor_sock.recv_into(int_buf, 8)
                 # client disconnect
-                if not line:
+                if should_be_eight < 8:
                     # If socket is done, we can't log
                     print('Client disconnect')
                     return
+                bytes_to_read = int.from_bytes(int_buf, "big")
```

Comment:
1. Use single quotes
2. Add size protection to be more robust against errors (e.g. if your max buffer is 4MB, check that size isn't over 4MB
3. Would also make the 8 byte header be 4 bytes magic number, 4 bytes size. This way you can statistically reduce reading garbage as the size to effectively nothing. If you read an invalid magic number, read the next 8 bytes (after magic) until you read a proper magic

Failure to do 2 and 3 will cause things to get stuck under error conditions